frame/contracts/src/storage.rs

```diff
@@ -75,15 +73,19 @@ where
 	/// `read`, this function also requires the `account` ID.
 	///
 	/// If the contract specified by the id `account` doesn't exist `Err` is returned.`
+	///
+	/// # Panics
+	///
+	/// Panics iff the `account` specified is not alive and in storage.
```

Comment:
I kind of prefer having a `Result<DispatchResult, ContractAbsentError>` than a panic, but I guess this is personal.

This makes the proof `.expect("...; qed")` more explicitly required IMO

Reply:
Yes me too. However, in this PR I added a non-panicing error (`StorageExhausted`). In order to have the `expect` at the caller side it would need to match over the `DispatchError` somehow to differentiate between the fatal and the non fatal one.

I am planning to remove all those expects soon. The plan is to load the contract info once and pass reference to those functions. Then we can get rid of all those proofs. This will be accomplished by #6439 and #6448.

Reply:
I mean you can still have a `Result<DispatchResult, ContractAbsentError>` to differentiate both, it is a bit a ugly, but safer
but I'm ok with current implementation especially if panic will be removed in future

Reply:
You are right. I did not think about nesting them.